.github/workflows/labeler.yml

```diff
@@ -16,3 +16,27 @@ jobs:
         sync-labels: 'true'
         dot: 'true'
         non-matching-label: 'no-match-files'
+  
+  external_pr_labeller:
+    name: Label External PR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Labeller Script
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: '.github'
```

Comment:
Looks like here you checkout the code from PR's context (due to pull_request_target trigger). Have you checked if it's possible in this setup to change the contents of .github/scripts/external_pr_labeller.py so that GitHub secrets are exposed? It doesn't look safe

Reply:
We can try to checkout labeler script from the target branch instead, but will it stop external PR's author from changing the workflow file itself in a harmful way? Not sure, also need to check

Reply:
The `.github` folder is checked out from the target branch context as it has the `pull_request_target` trigger, not from the PR. The default behaviour of the `actions/checkout` with `pull_request_target` is to checkout from the safe target branch context.
The changes in `.github/scripts/external_pr_labeller.py` via PRs will not be executed.

If I were to use `ref` in `actions/checkout`, only then would it checkout the unsafe PR code.

Reply:
Ah, I see, so GitHub eventually hardened the security to checkout in context of target repository by default. 

By the way, why not using https://github.com/openvinotoolkit/openvino/blob/master/.github/github_org_control/check_pr.py then?